packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -1145,59 +1147,68 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with Restoratio
     return ValueListenableBuilder<TextEditingValue>(
       valueListenable: _effectiveController,
       child: editableText,
-      builder: (BuildContext context, TextEditingValue? text, Widget? child) {
+      builder: (BuildContext context, TextEditingValue text, Widget? child) {
+        final bool hasText = text.text.isNotEmpty;
+        final String? placeholderText = widget.placeholder;
+        final Widget? placeholder = placeholderText == null
+          ? null
+          : Visibility(
```

Comment:
Before this change the placeholder would not be built at all if there was text, but now it is built regardless but hidden when there is text. This is so the input field does not shrink when the user starts entering text because now the placeholder is still there just hidden.

Reply:
Just wondering if I am understanding this change correctly.

Reply:
Yeah the placeholder will be kept in the widget tree to make sure the text field's height is always greater than or equal to the height of the first line of the placeholder, even when there's user text.